pkg/quay/quay_test.go

```diff
@@ -337,75 +626,92 @@ func TestQuayClient_GetTagsFromPage(t *testing.T) {
 					}
 				}
 
-				gock.New("https://quay.io/api/v1").
+				resp := gock.New("https://quay.io/api/v1").
 					MatchHeader("Authorization", "Bearer authtoken").
 					MatchHeader("Content-Type", "application/json").
 					Get(fmt.Sprintf("repository/%s/%s/tag/", org, repo)).
 					MatchParam("page", fmt.Sprintf("%d", page)).
-					Reply(200).
-					JSON(map[string]interface{}{
+					Reply(tc.statusCode)
+
+				if tc.expectedErr == "" {
+					resp.JSON(map[string]interface{}{
 						"tags":           mockTags,
 						"has_additional": tc.hasAdditional[page-1],
 					})
-				tags, hasAdditional, err := quayClient.GetTagsFromPage(org, repo, page)
-				if err != nil {
-					t.Errorf("error getting all tags from page, expected `nil`, got `%s`", err)
+				} else {
+					resp.JSON("{\"has_additional: false}")
 				}
-				if !reflect.DeepEqual(mockTags, tags) {
-					t.Errorf("tags are not the same, expected `%v`, got `%v`", mockTags, tags)
-				}
-				if hasAdditional != tc.hasAdditional[page-1] {
-					t.Errorf("hasAdditional is not the same, expected `%t`, got `%t`", tc.hasAdditional[page-1], hasAdditional)
+
+				tags, hasAdditional, err := quayClient.GetTagsFromPage(org, repo, page)
+				if tc.expectedErr == "" {
+					assert.NilError(t, err)
+
+					if !reflect.DeepEqual(mockTags, tags) {
+						t.Errorf("tags are not the same, expected `%v`, got `%v`", mockTags, tags)
+					}
+					if hasAdditional != tc.hasAdditional[page-1] {
+						t.Errorf("hasAdditional is not the same, expected `%t`, got `%t`", tc.hasAdditional[page-1], hasAdditional)
+					}
```

Comment:
`assert.DeepEqual` and `assert.Condition` might work here?

Reply:
Updated.